shared/login/index.js

```diff
@@ -16,26 +20,37 @@ export default class Login extends Component {
 
   static parseRoute (currentPath, uri) {
     // Fallback (for debugging)
-    let Form = () => <ErrorText currentPath={currentPath} />
-
-    switch (currentPath.get('path')) {
-      case 'root':
-        Form = () => <Intro/>
-        break
-      case 'signup':
-      case 'inviteCode':
-        Form = () => <InviteCode/>
-        break
-      case 'usernameAndEmail':
-        Form = () => <UsernameEmailForm/>
-        break
+    let Form = <ErrorText currentPath={currentPath} />
+
+    const path = currentPath.get('path')
+
+    const {component: Component, props} = currentPath.get('parseRoute') || {}
+    if (Component) {
+      Form = <Component {...props}/>
+    } else {
+      switch (path) {
+        case 'root':
+          Form = <Intro/>
+          break
+        case 'signup':
+        case 'inviteCode':
+          Form = <InviteCode/>
+          break
+        case 'usernameAndEmail':
+          Form = <UsernameEmailForm/>
+          break
+        case 'register':
+          Form = <Register />
+          break
+      }
     }
 
     return {
       componentAtTop: {
         component: Login,
+        hideBack: true,
         props: {
-          formComponent: Form
+          formComponent: () => Form
```

Comment:
minor :sheep: if this is no longer a react class can we lowercase it? to just `form`